lib/ebanq_api/client.rb

```diff
@@ -36,15 +36,18 @@ def initialize
     # * +values+ - values from request
     # Returns full body of response if no errors are found,
     # raise error otherwise.
-    def make_request(method, url, values = {})
+    def make_request(method, url, params = {})
+      path = "#{EbanqApi.base_url}/#{url}"
       response = if method == :get
-                   RestClient.get("#{EbanqApi.base_url}/#{url}", headers)
+                   RestClient.get(path, headers.merge!(params: params))
                  else
-                   RestClient.post("#{EbanqApi.base_url}/#{url}", values,
-                                   headers)
+                   RestClient.post(path, params, headers)
                  end
-      result = JSON.parse(response.body)
-      success?(result['code']) ? result['response'] : parse_failed(result)
+      if !response.body.empty?
+        process_response(response)
+      else
+        response.body
```

Comment:
In this case you return empty string.  It's better to return same response for all the cases, but with different structure in it. For example: if "result['response']" (above) gives us array of something then in empty case we should return empty array.

Reply:
After some changes I found that it will be no responses with empty body from this API(example of empty list response: {"page"=>1, "pages_count"=>0, "items"=>[]}) or [].
But I replaced empty string to empty hash just in case.